be/src/olap/task/engine_clone_task.cpp

```diff
@@ -96,176 +93,108 @@ Status EngineCloneTask::_do_clone() {
-            LOG(WARNING) << "failed to clone tablet. tablet_id:" << _clone_req.tablet_id
+    TTabletInfo tablet_info;
+    tablet_info.__set_tablet_id(_clone_req.tablet_id);
+    tablet_info.__set_replica_id(_clone_req.replica_id);
+    tablet_info.__set_schema_hash(_clone_req.schema_hash);
+    RETURN_IF_ERROR(StorageEngine::instance()->tablet_manager()->report_tablet_info(&tablet_info));
+    if (_clone_req.__isset.committed_version &&
+        tablet_info.version < _clone_req.committed_version) {
+        // if it is a new tablet and clone failed, then remove the tablet
+        // if it is incremental clone, then must not drop the tablet
+        if (is_new_tablet) {
+            // we need to check if this cloned table's version is what we expect.
+            // if not, maybe this is a stale remaining table which is waiting for drop.
+            // we drop it.
+            LOG(WARNING) << "begin to drop the stale tablet. tablet_id:" << _clone_req.tablet_id
                          << ", replica_id:" << _clone_req.replica_id
                          << ", schema_hash:" << _clone_req.schema_hash
                          << ", signature:" << _signature << ", version:" << tablet_info.version
                          << ", expected_version: " << _clone_req.committed_version;
-            // if it is a new tablet and clone failed, then remove the tablet
-            // if it is incremental clone, then must not drop the tablet
-            if (is_new_tablet) {
-                // we need to check if this cloned table's version is what we expect.
-                // if not, maybe this is a stale remaining table which is waiting for drop.
-                // we drop it.
-                LOG(WARNING) << "begin to drop the stale tablet. tablet_id:" << _clone_req.tablet_id
-                             << ", replica_id:" << _clone_req.replica_id
-                             << ", schema_hash:" << _clone_req.schema_hash
-                             << ", signature:" << _signature << ", version:" << tablet_info.version
-                             << ", expected_version: " << _clone_req.committed_version;
-                Status drop_status = StorageEngine::instance()->tablet_manager()->drop_tablet(
-                        _clone_req.tablet_id, _clone_req.replica_id, false);
-                if (drop_status != Status::OK() &&
-                    drop_status.precise_code() != OLAP_ERR_TABLE_NOT_FOUND) {
-                    // just log
-                    LOG(WARNING) << "drop stale cloned table failed! tablet id: "
-                                 << _clone_req.tablet_id;
-                }
-            }
-            status = Status::InternalError("Failed to clone tablet");
-        } else {
-            LOG(INFO) << "clone get tablet info success. tablet_id:" << _clone_req.tablet_id
-                      << ", schema_hash:" << _clone_req.schema_hash << ", signature:" << _signature
```

Comment:
In previous logic, _tablet_infos is only set in this logic, not in 233 logic. But in your code, _tablet_infos is always set.